packages/fxa-payments-server/server/lib/amplitude.js

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const {
+  GROUPS,
+  initialize,
+  mapBrowser,
+  mapFormFactor,
+  mapLocation,
+  mapOs,
+  mapUserAgentProperties,
+  toSnakeCase,
+} = require('../../../fxa-shared/metrics/amplitude.js');
+const config = require('../config');
+const amplitude = config.get('amplitude');
+const joi = require('joi');
+const logger = require('./logging/log');
+const ua = require('../../../fxa-shared/metrics/user-agent');
+const { version: VERSION } = require('../../package.json');
+let log, transform;
+
+const FUZZY_EVENTS = new Map([
+  [
+    // Emit events from the front-end as `amplitude.${GROUP}.${EVENT}`
+    /^amplitude\.([\w-]+)\.([\w-]+)$/,
+    {
+      group: group => GROUPS[group],
+      event: (group, event) => toSnakeCase(event),
+    },
+  ],
+]);
+
+// Ease testing by allowing mocks to be directly injected.
```

Comment:
I don't think we should have a special carve-out for testing here, or module-global vars for `transform` & `log`. Looks like they're only used in the function exported by the module. Can't we pass `transform` and `log` in as parameters from the code using this exported function - i.e. whether "real" or testing code?

Reply:
Oh, are tests for this code deferred for a follow-up? Can't find another module using this stuff https://github.com/mozilla/fxa/issues/2938

Reply:
Yeah, tests are deferred for the moment, but I felt weird about adding that injection spot. I'll remove it.